### DISCUSSION TITLE
Possible to connect to MSSQL via AD/NTLM from non-domain computer?

### DISCUSSION THREAD

**Discussion by k3davis** [dbeaver/dbeaver#16015] [Q&A] [answered]
#### System information: 
- Operating system (distribution) and version Win 11 Pro 21H2
- DBeaver version CE 22.0.0.202203061937
- Additional extensions none

#### Connection specification:
- Database name and version MS SQL Server 2019
- Driver name Microsoft JDBC Driver for SQL Server
- Do you use tunnels or proxies (SSH, SOCKS, etc)? no

#### Describe the problem you're observing:

Is it possible to establish a connection to an MS SQL database while not on a domain joined computer (e.g.. a home computer connected to the network via VPN) via some version of connection settings?

#### Steps to reproduce, if exist:

I have tried: Active Directory - Password, which returns `Login failed for user ''.` (same as #7165) and NTLM which returns `Login failed. The login is from an untrusted domain and cannot be used with Integrated authentication.`.

I can use other apps like SQL Server Management Studio or Azure Data Studio but need to do some gymnastics to get the app to run as the domain user, then I can use integrated auth. I was hoping there would be a way to connect by configuration instead of by the user the app itself is running as, which would be a lot less cumbersome.

**Comment by k3davis** [comment]
btw I don't really classify this as a "bug" but none of the other categories seemed fitting either. Please relabel as necessary.

**Comment by tilmanb** [comment]
A workaround would be to create a SQL Server Login with password on the SQL Server and map that login to the domain user in the database, then you can connect using username+password from dbeaver.
Also, if you can, please share the gymnastics you use to run SSMS as a domain user on a non-domain client.

I think integrated auth is meant to be integrated, so not configurable, because the OS is doing the validation.  See [Microsoft Docs](https://docs.microsoft.com/en-us/sql/relational-databases/security/choose-an-authentication-mode?view=sql-server-ver15) on this.

**Comment by Karlodun** [comment]
SQL Server windows login requires your PC to be in Domain. This is a security feature. Your PC doesn't need to be a windows PC, it can be Mac or Linux, but needs to be a member of domain.
Beyond timanb suggestion you could ask your admins to enable Kerberos configuration, but it would require certain level of domain integration anyways (config krb on PC):
https://docs.microsoft.com/en-us/sql/connect/jdbc/using-kerberos-integrated-authentication-to-connect-to-sql-server?view=sql-server-ver15

**Comment by k3davis** [comment, marked as answer]
@tilmanb I'm happy to share my workaround, but it's a little involved so I'm not sure the best place to do so. As an overview though, I use PowerShell do to something like this

```
Install-Module RunAs
ConvertFrom-SecureString (Get-Credential 'domain\user').Password
````

I copy the string returned from the second command and use it as a parameter in a new PowerShell script:

```
Import-Module RunAs
$mycreds = New-Object Management.Automation.PSCredential('domain\user', (ConvertTo-SecureString 'PASS-STRING'))
runas -netonly $mycreds "C:\path\to\program.exe"
```

Then when I execute the script, the app runs as the domain user, and I can use integrated auth. But you do have to update the script every time your password changes, which is probably more frequent than you would like, haha.

I'll look into the local-user-mapped-to-domain-user suggestion. I've tried Kerberos before but was never successful getting the configuration right so I could connect with it.

Anyway - seems the answer is "no" on my original question, so you can probably close this issue. Maybe the discussion will be helpful for someone else down the line.

**Reply by erdarun** [reply]
@k3davis Does same answer will work from MAC PowerShell? 

When I tried in MAC using Powershell, Got below error for final command **runas -netonly $mycreds "/Applications/DBeaver.app"**

> RunAs: Exception calling "Start" with "7" argument(s): "Value cannot be null. (Parameter 'path1')"

CC: @dbeaver-pm
